modules/ScreenNameLocalFormatter/src/main/java/com/liferay/formatter/validator/service/NewScreenNameValidator.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.formatter.validator.service;
+
+import com.liferay.formatter.keys.NewFormatterKeys;
+import com.liferay.petra.string.CharPool;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.security.auth.ScreenNameValidator;
+import com.liferay.portal.kernel.util.Html;
+import com.liferay.portal.kernel.util.Props;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author leonardo.ferreira
+ */
+@Component(
+	immediate = true, property = "service.ranking:Integer=100",
+	service = ScreenNameValidator.class
+)
+public class NewScreenNameValidator implements ScreenNameValidator {
+
+	public static final String POSTFIX = "postfix";
+
+	@Override
+	public String getAUIValidatorJS() {
```

Comment:
Some behavior in this class is similar to `DefaultScreenNameValidator`.
This method for example.

Would you be able to extend `DefaultScreenNameValidator` and override only the behavior you're interested in changing?
This would save some copy and paste.